pkg/shell/base_index.js

```diff
@@ -478,13 +478,13 @@ define([
          */
 
         /* Encode navigate state into a string */
-        function encode(state, sidebar) {
+        function encode(state, sidebar, with_root) {
```

Comment:
Lets add some documentation in a comment above. Why would someone ever call without ```with_root == true```?